bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/FloureonThermostatHandler.java

```diff
@@ -0,0 +1,260 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.broadlinkthermostat.internal.handler;
+
+import static org.openhab.binding.broadlinkthermostat.internal.BroadlinkThermostatBindingConstants.*;
+
+import java.io.IOException;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.mob41.blapi.FloureonDevice;
+import com.github.mob41.blapi.dev.hysen.AdvancedStatusInfo;
+import com.github.mob41.blapi.dev.hysen.BaseStatusInfo;
+import com.github.mob41.blapi.dev.hysen.SensorControl;
+import com.github.mob41.blapi.mac.Mac;
+import com.github.mob41.blapi.pkt.cmd.hysen.SetTimeCommand;
+
+/**
+ * The {@link FloureonThermostatHandler} is responsible for handling thermostats labeled as Floureon Thermostat.
+ *
+ * @author Florian Mueller - Initial contribution
+ */
+@NonNullByDefault
+public class FloureonThermostatHandler extends BroadlinkThermostatHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FloureonThermostatHandler.class);
+    private @Nullable FloureonDevice floureonDevice;
+
+    /**
+     * Creates a new instance of this class for the {@link FloureonThermostatHandler}.
+     *
+     * @param thing the thing that should be handled, not null
+     */
+    public FloureonThermostatHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Initializes a new instance of a {@link FloureonThermostatHandler}.
+     */
+    @Override
+    public void initialize() {
+        super.initialize();
+        if (host != null && macAddress != null) {
+            try {
+                blDevice = new FloureonDevice(host, new Mac(macAddress));
+                this.floureonDevice = (FloureonDevice) blDevice;
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Could not find broadlinkthermostat device at host" + host + "with MAC+" + macAddress + ": "
+                                + e.getMessage());
+            }
+            authenticate();
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command: {}", command.toFullString());
+
+        if (command == RefreshType.REFRESH) {
+            refreshData();
```

Comment:
Refresh is called for every channel. So if you have 10 channels this will be called 10 times. So you might want to build in some caching. You can do this for example with `ExpiringCache` on `AdvancedStatusInfo`. Where the actual call to the service is set in the cache. When getting the value you just call get on the cache and if the timeout has exceeded it will to a refresh. For example set the timeout to 3 seconds. So repeated calls due to refresh calls won't trigger a new call to the service.

Reply:
Cool idea. Applied.